### PR TITLE
Fix labels, structures, and reactions for C:3 H:2 species in CurranPentane

### LINKED ISSUE #526: Missing species in CurranPentane library
Entry 847 of the CurranPentane kinetic library has the reaction: "C3H2(S) + H <=> C3H2(S) + H" which was causing errors for the n-heptane model I was running.
Looking at the CurranPentane chemkin file, I believe the reaction was supposed to be "C3H2(S)+H<=>H2CCC(S)+H" but RMG is missing one of those species definitions. 

It has C3H2(S) but not H2CCC(S):
C3H2(S)
1 C u0 p0 c0 {2,D} {3,S} {4,S}
2 C u0 p0 c0 {1,D} {5,D}
3 H u0 p0 c0 {1,S}
4 H u0 p0 c0 {1,S}
5 C u0 p1 c0 {2,D}

This library might need to be reimported with both species defined separately.

### PR DESCRIPTION
**Pull request by mefuller**
Errors in label and structures for H2C3 isomers in the CurranPentane library are corrected:
- Previously omitted species H2CCC is added back
- C3H2 is updated to C#C[CH] from C=C=[C] (evolves from propargyl)
- All reactions for C3H2, C3H2(S), H2CCC(S), C3H2C (three-member ring) are checked and reassigned as needed
Closes #526

### CODE REVIEW

**Comment by rwest** [comment]
Looks believably right to me, but only looked at the changes and didn't try running anything.  So I approve the changes, but would like someone else (eg. @sevyharris who opened #526) to look at it also.

Are the CI test failures expected and understood and due to something else?

**Comment by sevyharris** [comment]
> Looks believably right to me, but only looked at the changes and didn't try running anything. So I approve the changes, but would like someone else (eg. @sevyharris who opened #526) to look at it also.
> 
> Are the CI test failures expected and understood and due to something else?

Those CI test failures are not expected as far as I can tell. I'll take a look.

File: trigger-rmg-tests.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ cd $TARGET_DIR
 
 # create a new branch in RMG-tests with the name equal to
 # the branch name of the tested RMG-database branch:
-if [ "$RMG_PY_BRANCH" == "master" ]
+if [ "$RMG_PY_BRANCH" == "main" ]
```
Comment (sevyharris) [inline trigger-rmg-tests.sh:33]: This is a good catch! There are far more rmgpydb-tests in RMG-tests than there should be, which is consistent with this bug. My only request is that if we're going to fix it here, we also have to fix it on line 45.

**Review by sevyharris — COMMENTED** [review]
This looks good, but I want to fix the other 'master' to 'main' typo in trigger-rmg-tests.sh before I run my tests. If you can't get to it tonight, I'll take care of it and this should be ready to go tomorrow.

File: trigger-rmg-tests.sh (hunk 1 above)
Comment (mefuller) [reply trigger-rmg-tests.sh:33]: Good catch - updated

**Comment by sevyharris** [comment]
Great! I'll run my tests and approve it when they're done.

**Review by sevyharris — APPROVED** [review]
Looks good! I ran a minimal example to check that it works, and then I also ran my n-heptane model that uses the CurranPentane library. It works as expected and fixes the original bug I described. I would still wait to see how it does on the regression tests before merging. It should pass at least a few of them, but if it doesn't, I think the issue is more likely with RMG-tests than this PR. Thanks for all the good work!

**Comment by mefuller** [comment]
The failed test here appears to be `
FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/RMG-tests/RMG-tests/code/testing/input/transport/groups/ring.py'` which is unrelated to this PR

**Comment by sevyharris** [comment]
> The failed test here appears to be ` FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/RMG-tests/RMG-tests/code/testing/input/transport/groups/ring.py'` which is unrelated to this PR

Yes, this error showed up in at least one other unrelated RMG-database PR (vinyl_adsorption_correction), so I think it's safe to say the issue is in RMG-tests. This PR looks good to me and I recommend @rwest merge the changes. I'll start looking into the issues with RMG-tests.

**Review by alongd — APPROVED** [review]
Looking good, thanks for the fixes!

**Pull request merged**